File: api-crypto/crypto/csharp/src/alg/MGF.cs
using System;
using System.Linq;
using System.Reflection;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.asn.algorithms;

//todo Annotation!
//@ApiClass

namespace tr.gov.tubitak.uekae.esya.api.crypto.alg
{
    /**
     * 
     * Mask Generation Function.
     *
     * @author ayetgin
     */
    public class MGF
    {
        public static readonly MGF MGF1 = new MGF(_algorithmsValues.id_mgf1, "MGF1");

        private readonly int[] mOID;
        private readonly String mName;

        private MGF(int[] aOID, String aName)
        {
            mName = aName;
            mOID = aOID;
        }

        public int[] getOID()
        {
            return mOID;
        }

        public String getName()
        {
            return mName;
        }

        public static MGF fromAlgorithmIdentifier(EAlgorithmIdentifier aAlgorithmIdentifier)
        {
            foreach (FieldInfo fieldInfo in typeof (MGF).GetFields())
            {
                MGF alg = (MGF) fieldInfo.GetValue(null);
                if (alg.getOID().SequenceEqual(aAlgorithmIdentifier.getObject().algorithm.mValue))
                    return alg;
            }
            return null;
        }

        public static MGF fromOID(int[] aOID)
        {
            foreach (FieldInfo fieldInfo in typeof (MGF).GetFields())
            {
                MGF alg = (MGF) fieldInfo.GetValue(null);
                if (aOID.SequenceEqual(alg.getOID()))
                    return alg;
            }
            return null;
        }

        public static MGF fromName(String algName)
        {
            foreach (FieldInfo fieldInfo in typeof (MGF).GetFields())
            {
                MGF alg = (MGF) fieldInfo.GetValue(null);
                if (alg.getName().Equals(algName, StringComparison.OrdinalIgnoreCase))
                    return alg;
            }

            return null;
        }
    }
}

File: api-crypto/crypto/src/tr/gov/tubitak/uekae/esya/api/crypto/alg/MGF.java
package tr.gov.tubitak.uekae.esya.api.crypto.alg;

import tr.gov.tubitak.uekae.esya.api.asn.x509.EAlgorithmIdentifier;
import tr.gov.tubitak.uekae.esya.asn.algorithms._algorithmsValues;

import java.util.Arrays;

/**
 * Mask Generation Function.
 *
 * @author ayetgin
 */

public enum MGF
{
    MGF1 (_algorithmsValues.id_mgf1, "MGF1");

    private int[] mOID;
    private String mName;

    private MGF(int[] aOID, String aName)
    {
        mOID = aOID;
        mName = aName;
    }

    public int[] getOID()
    {
        return mOID;
    }

    public String getName()
    {
        return mName;
    }

    public static MGF fromAlgorithmIdentifier(EAlgorithmIdentifier aAlgorithmIdentifier){
        for (MGF alg : MGF.values()) {
              if (Arrays.equals(aAlgorithmIdentifier.getAlgorithm().value, alg.getOID()))
                  return alg;
        }
        return null;
    }

    public static MGF fromOID(int[] aOID){
        for (MGF alg : MGF.values()) {
              if (Arrays.equals(aOID, alg.getOID()))
                  return alg;
        }
        return null;
    }

    public static MGF fromName(String algName){
        for (MGF alg : values()) {
            if (alg.getName().equalsIgnoreCase(algName))
                return alg;
        }

        return null;
    }

}
